src/wizard/wizard-step.component.js

```diff
@@ -0,0 +1,325 @@
+ * @param {boolean} substeps Sets whether this step has substeps
+ * @param {boolean=} nextEnabled Sets whether the next button should be enabled when this step is first displayed
+ * @param {boolean=} prevEnabled Sets whether the back button should be enabled when this step is first displayed
+ * @param {string=} nextTooltip The text to display as a tooltip on the next button
+ * @param {string=} prevTooltip The text to display as a tooltip on the back button
+ * @param {boolean=} wzDisabled Disables the wizard when this page is shown
+ * @param {boolean} okToNavAway Sets whether or not it's ok for the user to leave this page
+ * @param {boolean} allowClickNav Sets whether the user can click on the numeric step indicators to navigate directly to this step
+ * @param {string=} description The step description (optional)
+ * @param {object} wizardData Data passed to the step that is shared by the entire wizard
+ * @param {function()=} onShow The function called when the wizard shows this step
+ * @param {boolean=} showReview Indicates whether review information should be displayed for this step when the review step is reached
+ * @param {boolean=} showReviewDetails Indicators whether the review information should be expanded by default when the review step is reached
+ * @param {string=} reviewTemplate The template that should be used for the review details screen
+ */
+angular.module('patternfly.wizard').component('pfWizardStep', {
+  require: {
+    wizard: '^pfWizard'
+  },
+  transclude: true,
+  bindings: {
+    stepTitle: '@',
+    stepId: '@',
+    stepPriority: '@',
+    substeps: '=?',
+    nextEnabled: '<?',
+    prevEnabled: '<?',
+    nextTooltip: '<?',
+    prevTooltip: '<?',
+    disabled: '@?wzDisabled',
+    okToNavAway: '<?',
+    allowClickNav: '<?',
+    description: '@',
+    wizardData: '=',
+    onShow: '=?',
+    showReview: '@?',
+    showReviewDetails: '@?',
+    reviewTemplate: '@?'
+  },
+  templateUrl: 'wizard/wizard-step.html',
+  controller: function ($timeout) {
+    'use strict';
+
+    var firstRun = true,
+      ctrl = this;
+
+    var stepIdx = function (step) {
+      var idx = 0;
+      var res = -1;
+      angular.forEach(ctrl.getEnabledSteps(), function (currStep) {
+        if (currStep === step) {
+          res = idx;
+        }
+        idx++;
+      });
+      return res;
+    };
+
+    var unselectAll = function () {
+      //traverse steps array and set each "selected" property to false
+      angular.forEach(ctrl.getEnabledSteps(), function (step) {
+        step.selected = false;
+      });
+      //set selectedStep variable to null
+      ctrl.selectedStep = null;
+    };
+
+    var stepByTitle = function (titleToFind) {
+      var foundStep = null;
+      angular.forEach(ctrl.getEnabledSteps(), function (step) {
+        if (step.stepTitle === titleToFind) {
+          foundStep = step;
+        }
+      });
+      return foundStep;
+    };
+
+    ctrl.steps = [];
+    ctrl.context = {};
```

Comment:
I'd prefer to see all initialization together rather than scattered between function definitions, etc.